### PR TITLE
Add date and time type in input

### PR DESCRIPTION
**Pull request by abdolian**

### CODE REVIEW

File: src/FluentCMS.Web.UI/Components/Core/Form/Input.razor [hunk 1]
```diff
@@ -6,8 +6,8 @@
         <div class=@this.ClassName("InputIconWrapper")>
             <Icon Name="@IconName" />
         </div>
-        <input 
-            type="text" 
+        <input
+            type=@this.Type.ToString().Replace("_", "-")
```
Comment (ParsaGachkar) [inline src/FluentCMS.Web.UI/Components/Core/Form/Input.razor:10]: We have a `FromPascalCaseToKebabCase()` extension method for this logic. can we use that?

File: src/FluentCMS.Web.UI/Components/Enums/InputTypeEnum.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+namespace FluentCMS.Web.UI.Components;
+
+public enum InputTypeEnum
+{
+    Date,
+    Datetime_Local,
```
Comment (ParsaGachkar) [inline src/FluentCMS.Web.UI/Components/Enums/InputTypeEnum.cs:6]: Dont use `_` to Seperate Words Just use PascalCase and `FromPascalCaseToKebabCase()` will handle the rest

**Review by ParsaGachkar — COMMENTED** [review]
the logic is fine, just some cleanup required!

**Comment by TheHadiAhmadi** [comment]
I think it is better to have multiple components instead of one `Input` component.

`TextInput` extends `InputBaseComponent<string>`
`DateInput` extends `InputBaseComponent<Date>`
`SelectInput` extends `InputBaseComponent<string>`
`MultiSelectInput` extends `InputBaseComponent<string[]>`
`NumberInput` extends `InputBaseComponent<int>`
...

we can change type of `Value` prop based on component type.

@parsaGachkar @abdolian @babakhani @pournasserian 
What do you think?

**Review by ParsaGachkar — APPROVED** [review]

**Pull request merged**